docs/dev/control-eval.md

```diff
@@ -0,0 +1,62 @@
+# What happens when a profile file is loaded
+
+## Consult with Harry Tuttle
+
+[He's not from Central Services or anything.](https://youtu.be/VRfoIyx8KfU?t=2m41s)
+
+## Tips
+
+* In the early days of InSpec / ServerSpec, controls were called "rules".  Throughout various places in the code, the word "rule" is used to mean "control".  Make the mental subsitution.
+* InSpec supports reading profiles from tarballs, local files, git repos, etc.  So, don't count on local file reading; instead it uses a special source reader to obtain the contents of the files.
+
+## The basics of the stack
+
+    #5  Inspec::Profile.collect_tests(include_list#Array) at lib/inspec/profile.rb:167
+    #4  Hash.each at lib/inspec/profile.rb:167
+    #3  block in Inspec::Profile.block in collect_tests(include_list#Array) at lib/inspec/profile.rb:170
+    #2  Inspec::ProfileContext.load_control_file(*args#Array) at lib/inspec/profile_context.rb:141
+    #1  Inspec::ProfileContext.control_eval_context at lib/inspec/profile_context.rb:58
+    #0  #<Class:Inspec::ControlEvalContext>.create(profile_context#Inspec::ProfileContext, resources_dsl#Module) at lib/inspec/control_eval_context.rb:41
+
+## A profile context is created
+
+Like many things in InSpec core, a profile context is an anonymous class. (verify)
```

Comment:
Could we remove `(verify)`?

Reply:
This is a document for us, the inspec team.  I wouldn't expect it to be consumed publicly, or if so, only by people trying to work on the guts of inspec. Much lower standdards - the goal is to capture tribal knowledge.  It's incomplete, both structurally and content-wise.  Some areas might even be factually incorrect.  Here, I'm just tempering my certainty; the particular snipe-hunt I was on didn't entail me looking very hard at the profile context.  Next time someone needs to know this sort of tribal knowledge, perhaps they can shore it up a bit more, if their research leads them in that direction.

Reply:
Ah, that makes sense. I suppose leaving those sorts of things in are fine then.